documentation/Using-the-SDK/Advanced/Disposable-Extensions.md

```diff
@@ -1,37 +1,41 @@
 # Abstract
 
-This document outlines making your Custom Data Sources and Extensions
-disposable.
+This document outlines how to make a `Custom Data Source` or SDK
+extension disposable.
 
 # Motiviation
 
-Your extension manages items that require cleanup, and you would like to be
-able to implement `IDisposable` and have the SDK's runtime (Runtime) make sure
-that your extension is disposed when executioin is complete.
+When your extension manages items that require cleanup, it would be useful to
+implement `IDisposable` and have the SDK's runtime make sure that your
+extension cleans up its resources.
 
 # Usage
 
-In order for the Runtime to invoke your `IDisposable` implementation, all you
-need to do is make sure your extension implmenets `IDisposable`.
+In order for the SDK runtime to invoke your `IDisposable` implementation, you
+only need to make sure your extension implmenets `IDisposable`.
 
 Users may add the `IDisposable` interface to types implementing the following
-interfaces, whether they are implemented directly or via one of the SDK 
-provided base classes:
+interfaces:
 - `ICustomDataSource`
 - `ICustomDataProcessor`
 - `ISourceDataCooker`
 - `ICompositeDataCookerDescriptor`
 - `IDataProcessor`
 
 Example:
```

Comment:
to be consistent with other examples this can probably be changed to "For example, you can do:"